### PR TITLE
Fix azure list-objects not to list minio.sys.tmp/

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Because of incorrect comparison, we were listing `minio.sys.tmp/` in non-recursive listing output

## How Has This Been Tested?

manual and functional tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :x: [more...](#c026f400-077a-4785-93c5-2406779e95ea) |
<a name="c026f400-077a-4785-93c5-2406779e95ea"></a>
### 5895-996987b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:31654
ACCESS_KEY:      minio
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp a5f1a9cd1479:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... done in 3 seconds
(2/13) Running aws-sdk-java tests ... done in 15 seconds
(3/13) Running aws-sdk-php tests ... done in 48 seconds
(4/13) Running aws-sdk-ruby tests ... done in 18 seconds
(5/13) Running awscli tests ... done in 6 minutes and 3 seconds
(6/13) Running mc tests ... done in 1 minutes and 59 seconds
(7/13) Running minio-dotnet tests ... done in 47 seconds
(8/13) Running minio-go tests ... done in 12 seconds
(9/13) Running minio-java tests ... done in 51 seconds
(10/13) Running minio-js tests ... FAILED in 59 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-a563fcba-9e05-493d-930b-4d31a1602f1f, events: s3:ObjectCreated:*",
  "duration": 8391,
  "status": "FAIL",
  "error": "Error: notification not received at client.removeObject.err (test/functional-tests.js:1034:43) at ClientRequest.<anonymous> (node_modules/minio/dist/main/minio.js:420:9) at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:478:21) at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23) at Socket.socketOnData (_http_client.js:367:20) at readableAddChunk (_stream_readable.js:176:18) at Socket.Readable.push (_stream_readable.js:134:10) at TCP.onread (net.js:559:20)"
}

Executed 9 out of 13 tests successfully.
```

File: cmd/gateway/azure/gateway-azure.go [hunk 1]
```diff
@@ -572,7 +572,7 @@ func (a *azureObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
 		}
 
 		for _, blobPrefix := range resp.BlobPrefixes {
-			if prefix == minio.GatewayMinioSysTmp {
+			if blobPrefix == minio.GatewayMinioSysTmp {
 				// We don't do strings.HasPrefix(blob.Name, minio.GatewayMinioSysTmp) here so that
 				// we can use tools like mc to inspect the contents of minio.sys.tmp/
 				// It is OK to allow listing of minio.sys.tmp/ in non-recursive mode as it aids in debugging.
```
Comment (poornas) [inline cmd/gateway/azure/gateway-azure.go:578]: as per this comment, looks like it was deliberately left in so as to inspect minio.sys contents. Is this no longer required? If so, this comment should also be cleaned up.

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/gateway/azure/gateway-azure.go (hunk 1 above)
Comment (krishnasrinivas) [reply cmd/gateway/azure/gateway-azure.go:578]: @poornas the comment says (unlike the recursive listing) we don't do strings.HasPrefix (here we instead do `==`)

this is to support this output:
```
krishna@escape:~$ mc ls myazure/minio-js-test-20edb793-1e6b-4325-9f28-41070871be01/
[2018-05-07 16:49:55 PDT]    10B 259896e7-259e-4d36-bcd3-3f6693349218
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.0
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.1
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.2
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.3
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.4
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.5
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.6
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.7
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.8
[2018-05-07 16:49:51 PDT]     1B miniojsPrefix.9
[2018-05-07 18:52:20 PDT]     0B 1/
krishna@escape:~$ 
```

notice that there is no minio.sys.tmp in this output

but using mc we can inspect inside minio.sys.tmp:
```
krishna@escape:~$ mc ls myazure/minio-js-test-20edb793-1e6b-4325-9f28-41070871be01/minio.sys.tmp/multipart/v1/
[2018-05-07 18:52:58 PDT]     0B 161d957ee6363c88.1343f38db3f7a65f950f2c4967832cd8513bcb8aaaf7ba74782396e32405001b/
[2018-05-07 18:52:58 PDT]     0B 58d605dcea43eee2.8a1abc59b53c058c8e350ab98cec3ca6e7a4fa3375b1a161ffee0760a1cbac49/
[2018-05-07 18:52:58 PDT]     0B 8d74c1e33fae3ea4.4f094ed9ffbe187425ebd84ebd129683d55550943dd31f2bee38414c7e59661c/
krishna@escape:~$ 
```

Comment (poornas) [reply cmd/gateway/azure/gateway-azure.go:578]: ok, missed that.

**Review by poornas — APPROVED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**